src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java

```diff
@@ -0,0 +1,201 @@
+package org.folio.service.manager.inputdatamanager;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.shareddata.LocalMap;
+import org.folio.dao.impl.JobExecutionDaoImpl;
+import org.folio.rest.jaxrs.model.ExportRequest;
+import org.folio.rest.jaxrs.model.FileDefinition;
+import org.folio.service.manager.exportmanager.ExportManager;
+import org.folio.service.manager.exportmanager.ExportPayload;
+import org.folio.service.manager.inputdatamanager.datacontext.InputDataContext;
+import org.folio.service.manager.inputdatamanager.reader.SourceReader;
+import org.folio.service.manager.exportresult.ExportResult;
+import org.folio.service.upload.definition.FileDefinitionService;
+import org.folio.spring.SpringContextUtil;
+import org.folio.util.OkapiConnectionParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.lang.invoke.MethodHandles;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static io.vertx.core.Future.succeededFuture;
+
+/**
+ * Acts a source of a uuids to be exported.
+ */
+@SuppressWarnings({"java:S1172", "java:S125"})
+@Service
+class InputDataManagerImpl implements InputDataManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private static final int POOL_SIZE = 1;
+  private static final String INPUT_DATA_LOCAL_MAP_KEY = "inputDataLocalMap";
+  private static final String SHARED_WORKER_EXECUTOR_NAME = "input-data-manager-thread-worker";
+  private static final String TIMESTAMP_PATTERN = "yyyyMMddHHmmss";
+  private static final String DELIMITER = "-";
+  private static final int BATCH_SIZE = 1;
+
+
+  @Autowired
+  private SourceReader sourceReader;
+  @Autowired
+  private JobExecutionDaoImpl jobExecutionDao;
+  @Autowired
+  private FileDefinitionService fileDefinitionService;
+  @Autowired
+  private Vertx vertx;
+
+  private WorkerExecutor executor;
+  private LocalMap<String, InputDataContext> inputDataLocalMap;
```

Comment:
For your information: Here we use Vertx Shared Data API that allows us to safely share data between different parts of the application. See https://vertx.io/docs/vertx-core/java/#_using_the_shareddata_api